### PR TITLE
fix installation of Python bindings for recent Arrow easyconfigs by add `--no-build-isolation` to `pip install` command

### LINKED ISSUE #18327: Failure to build pyarrow wheel due to Cython incompatibility during Arrow install
Building `Arrow-11.0.0-gfbf-2022b.eb` I am hitting the error below.

The error looks to be related to the changes in Cython 3, which was released yesterday. However, I am confused as to why we're not using the Cython (0.29.32) in EB Python.

Adding the following to the Arrow easyconfig fixes it, further pointing at Cython 3 being the issue (see https://github.com/bear-rsg/bear-eb/blob/b597bbce58316d85d0d00dc1a95047c5fe9152e3/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb).
```
local_install_pyarrow_cmds += "sed -i 's/cython >= 0.29.22/cython >= 0.29.22,<3/g' pyproject.toml && "
```

Error:
```
      [  1%] Compiling Cython CXX source for lib...

      Error compiling Cython file:
      ------------------------------------------------------------
      ...

      cdef extern from "Python.h":
          int PySlice_Check(object)


      cdef int check_status(const CStatus& status) nogil except -1
                                                                  ^
      ------------------------------------------------------------

      pyarrow/lib.pxd:67:60: The keyword 'nogil' should appear at the end of the function signature line. Placing it before 'except' or 'noexcept' will be disallowed in a future version of Cython.


      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      ArrowIOError = IOError


      # This function could be written directly in C++ if we didn't
      # define Arrow-specific subclasses (ArrowInvalid etc.)
      cdef int check_status(const CStatus& status) nogil except -1:
                                                                  ^
      ------------------------------------------------------------

      pyarrow/error.pxi:82:60: The keyword 'nogil' should appear at the end of the function signature line. Placing it before 'except' or 'noexcept' will be disallowed in a future version of Cython.

      Error compiling Cython file:
      ------------------------------------------------------------
      ...
                  raise ArrowException(message)


      # This is an API function for C++ PyArrow
      cdef api int pyarrow_internal_check_status(const CStatus& status) \
              nogil except -1:
                             ^
      ------------------------------------------------------------

      pyarrow/error.pxi:143:23: The keyword 'nogil' should appear at the end of the function signature line. Placing it before 'except' or 'noexcept' will be disallowed in a future version of Cython.

      gmake[2]: *** [CMakeFiles/lib_pyx.dir/build.make:71: CMakeFiles/lib_pyx] Error 1
      gmake[1]: *** [CMakeFiles/Makefile2:146: CMakeFiles/lib_pyx.dir/all] Error 2
      gmake: *** [Makefile:136: all] Error 2
      error: command '/rds/bear-apps/2022b/EL8-ice/software/CMake/3.24.3-GCCcore-12.2.0/bin/cmake' failed with exit code 2
```

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

fixes #18327

### CODE REVIEW

File: easybuild/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb [hunk 1]
```diff
@@ -53,7 +53,8 @@ local_install_pyarrow_cmds += "cd %(builddir)s/*arrow-%(version)s/python && expo
 local_install_pyarrow_cmds += "sed -i 's/numpy==[0-9.]*/numpy/g' pyproject.toml && "
 local_install_pyarrow_cmds += "Python3_ROOT_DIR=$EBROOTPYTHON "
 local_install_pyarrow_cmds += "PYARROW_CMAKE_OPTIONS='-DZSTD_LIB=$EBROOTZSTD/lib/libzstd.%s ' " % SHLIB_EXT
-local_install_pyarrow_cmds += "PYARROW_WITH_DATASET=1 PYARROW_WITH_PARQUET=1 pip install --prefix %(installdir)s ."
+local_install_pyarrow_cmds += "PYARROW_WITH_DATASET=1 PYARROW_WITH_PARQUET=1 "
+local_install_pyarrow_cmds += "pip install --prefix %(installdir)s --no-build-isolation ."
```
Comment (jfgrimm) [inline easybuild/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb:57]: should we also make this `python -m pip install ...` instead of just `pip install`?

Comment (verdurin) [reply easybuild/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb:57]: @branfosj thoughts on the suggestion from @jfgrimm?

Comment (branfosj) [reply easybuild/easyconfigs/a/Arrow/Arrow-11.0.0-gfbf-2022b.eb]: 2f61086

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3142.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/165c70be17db05a8fa375427010fdfc4 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @branfosj!

**Comment by boegel** [comment]
@branfosj You missed a spot, see #18432

**Pull request merged**